src/pages/index.js

```diff
@@ -8,80 +8,172 @@ import { initialCards, cardData, options , cardSelector } from "../utils/constan
-export const popupImage = imagePreview.querySelector(".modal__preview-image");
-export const popupImageTitle = imagePreview.querySelector(".modal__caption");
-
-const userInfo = new UserInfo(profileTitle, profileDescription);
-
 const editFormValidator = new FormValidator(options, profileEditForm);
 editFormValidator.enableValidation();
 
 const addCardValidator = new FormValidator(options, addForm)
 addCardValidator.enableValidation();
 
-const imagePreviewPopup = new PopupWithImage("#preview_modal");
+const avatarValidator = new FormValidator(options, avatarForm)
+avatarValidator.enableValidation();
+
+const imagePreviewPopup = new PopupWithImage("#preview-modal");
 imagePreviewPopup.setEventListeners();
 
-function handleImageClick(data) {
-  imagePreviewPopup.open(data);
-};
+const confirmationPopup = new PopupWithConfirmation({popupSelector: options.deletePopup});
+confirmationPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#avatar-modal", handleAvatarChange);
+avatarPopup.setEventListeners();
 
+const api = new Api({
+  url: "https://around-api.en.tripleten-services.com/v1/",
+  headers: {
+    authorization: "0d8f734d-caf1-45e3-b9d3-764b4099955a",
+    "Content-Type": "application/json"
+  }
+}); 
+
+let cardSection;
+
+const renderCard = (data) => {
+  const card = createCard(data);
+  cardSection.addItem(card);
+};
 
-function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick)
-  return card.getView()
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData);
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      options.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const userInfo = new UserInfo(
+  profileTitle, 
+  profileDescription,
+  profileAvatar
+  );
+
+function createCard(data) {
+  const cardElement = new Card(
+    {
+      data,
+      handleImageClick: (imgData) => {
+        imagePreviewPopup.open(imgData);
+      },
+      handleDeleteClick: () => {
+        confirmationPopup.open(() => {
+          confirmationPopup.renderLoading(true);
+          api.deleteCard(data._id)
+            .then(() => {
+              cardElement.removeCard();
+              confirmationPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmationPopup.renderLoading(false));
+        });
+      },
+      handleLikeClick: () => {
+        if (cardElement.isLiked()) {
+          api
+            .removeLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        } else {
+          api
+            .addLike(cardElement._cardId)
```

Comment:
`_cardId`

Please, delete the **low dash** from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.